crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -145,19 +146,27 @@ impl<'src> Parser<'src> {
                     &lhs,
                 );
 
-                #[allow(deprecated)]
-                Box::new(Expr::Invalid(ast::ExprInvalid {
-                    value: self.src_text(lhs.range()).into(),
+                // In case it's not a valid LHS pattern, we'll use an empty `Expr::Name`
+                // to indicate that.
+                Box::new(Expr::Name(ast::ExprName {
+                    id: String::new(),
+                    ctx: ExprContext::Invalid,
                     range: lhs.range(),
                 }))
             };
 
             let rhs_pattern = self.parse_match_pattern_lhs();
             let rhs_value = if let Pattern::MatchValue(rhs) = rhs_pattern {
-                if !rhs.value.is_literal_expr() {
+                if !matches!(
+                    &*rhs.value,
+                    Expr::NumberLiteral(ast::ExprNumberLiteral {
+                        value: ast::Number::Complex { .. },
+                        ..
+                    })
+                ) {
```

Comment:
In a pattern `+` / `-` is always used for complex numbers. There can't be binary operations in a pattern.